### PR TITLE
Added buttons and support for changing to a tiled video layout.

### PR DESCRIPTION
**Pull request by Peterede**
Allow the participants in a call to switch the video layout

### CODE REVIEW

**Comment by nickvergessen** [comment]
Hmm, this PR conflicts heavily with our plan for a new UI:
https://github.com/nextcloud/spreed/issues/1056

Also @danxuliu is in the moment changing the WebRTC lib, to Adapter.JS, so I guess after that it won't apply cleanly anymore.

File: templates/index-public.php [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 	<div id="app-content-wrapper">
 		<button id="video-fullscreen" class="icon-fullscreen icon-white icon-shadow public" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Fullscreen (f)')) ?>"></button>
+		<button id="video-tiled" class="icon-videotiled icon-white icon-shadow hidden" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('toggle Tile view')) ?>"></button>
```
Comment (nickvergessen) [inline templates/index-public.php]:
```suggestion
		<button id="video-tiled" class="icon-videotiled force-icon-white-in-call icon-shadow hidden" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Toggle tile view')) ?>"></button>
```

File: templates/index.php [hunk 2]
```diff
@@ -27,6 +27,7 @@
 
 	<div id="app-content-wrapper">
 		<button id="video-fullscreen" class="icon-fullscreen icon-white icon-shadow hidden" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Fullscreen (f)')) ?>"></button>
+		<button id="video-tiled" class="icon-videotiled icon-white icon-shadow hidden" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Toggle Tile view')) ?>"></button>
```
Comment (nickvergessen) [inline templates/index.php]:
```suggestion
		<button id="video-tiled" class="icon-videotiled force-icon-white-in-call icon-shadow hidden" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Toggle tile view')) ?>"></button>
```

File: css/style.scss [hunk 3]
```diff
@@ -1095,3 +1095,63 @@ input[type="password"] {
...
+
+.icon-videolarge {
+    background-image: url('../img/videolarge.svg?v=1');
+}
+
+.icon-videotiled {
+    background-image: url('../img/videotiled.svg?v=1');
+}
+
+.videoLarge {
```
Comment (nickvergessen) [inline css/style.scss]: remove empty block

File: css/style.scss [hunk 4]
```diff
@@ -1095,3 +1095,63 @@ input[type="password"] {
 		filter: none !important;
 	}
 }
+
+.icon-videolarge {
```
Comment (nickvergessen) [inline css/style.scss]: Use the following instead:
```css
@include icon-black-white('menu-people', 'spreed', 1);
```
This makes sure it works correctly in all available color modes of Nextcloud.
Afterwards add it to the exception block in:
https://github.com/nextcloud/spreed/blob/339b1dfc05e3ff01c89f58757febb7a7fa97e887/css/style.scss#L323-L326

To make sure it's always white in the call screen.

**Comment by danxuliu** [comment]
>  Also @danxuliu is in the moment changing the WebRTC lib, to Adapter.JS, so I guess after that it won't apply cleanly anymore.

Indeed. In any case please note that the first step is refactoring the UI, so once that is done it would be possible to adjust this pull request; it will not be needed to wait until the full change is done.

A different issue is the conflict with #1056, though.

**Comment by nickvergessen** [comment]
Closing as per #1056

**Pull request closed without merging**